### ISSUE TITLE
Should ExternC functions be nothrow?

### ISSUE THREAD

**Issue by chloekek** [D-Programming-Deimos/openssl#45]
It is especially important that callbacks, such as those passed to ERR_print_errors_cb, do not throw exceptions. The compiler can enforce this if the callback has the nothrow attribute. Should we add this attribute to the library in the appropriate places?

**Comment by carblue** [comment]
Yes, absolutely. (answering without having looked into the code, just from my experience with callback functions used by C originating binaries). But enforcing nothrow abruptly may break a lot of code that doesn't throw but that isn't nothrow, thus there may be a need for a gentle migration (if there is a gentle way at all).
